cmake/Modules/OpmCompile.cmake

```diff
@@ -13,11 +13,13 @@ macro (opm_compile opm)
   # all public header files are together with the source. prepend our own
   # source path to the one of the dependencies so that our version of any
   # ambigious paths are used.
-  set (${opm}_INCLUDE_DIR "${PROJECT_SOURCE_DIR}")
-  set (${opm}_INCLUDE_DIRS ${${opm}_INCLUDE_DIR} ${${opm}_INCLUDE_DIRS})
+  # set (${opm}_INCLUDE_DIR "${PROJECT_SOURCE_DIR}")
+  # set (${opm}_INCLUDE_DIRS ${${opm}_INCLUDE_DIR} ${${opm}_INCLUDE_DIRS})
 
   # create this library, if there are any compilation units
-  include_directories (${${opm}_INCLUDE_DIRS})
+  # include_directories (${${opm}_INCLUDE_DIRS})
+  include_directories("${PROJECT_SOURCE_DIR}")
+  include_directories (SYSTEM ${${opm}_INCLUDE_DIRS})
```

Comment:
I'm seeing similar symptoms.  In my most common setup I have a separate installation directory for each build configuration (tuple of `CMAKE_BUILD_TYPE`, target Dune version (currently 2.3.1 and 2.4.1) and MPI-vs-sequential) and downstream modules are supposed to pick up the upstream modules from this per-configuration install directory and `make install` their outputs into it too.

When building the (not yet advertised) module [opm-flowdiagnostics](https://github.com/OPM/opm-flowdiagnostics), this PR (#202) removes the `-I` argument that points to the install directory.  As the module in question depends on opm-common (and Boost) only it cannot get its include directories transitively through other modules.  Reverting #202 locally restores the requisite `-I` argument.  I am happy to leave the situation as is locally but I would like to eventually understand what's actually happening.